### PR TITLE
Uppdated the model to output track and a few other datapoints.

### PR DESCRIPTION
**Pull request by Fberbari**
The model now produces everything sensor fusion would produce, so we've got a full software-in-the-loop simulation now.

### CODE REVIEW

**Comment by Fberbari** [comment]
Note, I gotta do a final test or 2 before I'm ready for merge. I put this up so the review could begin, but I'm realizing there's not much to review... sigh.

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+#include <fstream>
+#include <stdio.h>
+#include <string>
+#include <cmath>
+
+/***********************************************************************************************************************
+ * Prototypes
+ **********************************************************************************************************************/
+
+constexpr float m_pi = 3.14159;
```
Comment (NixonCTChan) [inline Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:13]: Are we gonna do anything about these declarations of Pi anytime soon?

**Review by NixonCTChan — COMMENTED** [review]
Everything looks pretty good! Can't really review the simulink model, but all the other changes seem ok!

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp (hunk 1 above)
Comment (Fberbari) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:13]: yes, I'm gonna treat that as my next task, but I'm gonna be making a dedicated PR for it.

File: Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h [hunk 2]
```diff
@@ -117,7 +117,9 @@ typedef struct _ssSolverInfo_tag {
     int_T       ignoredZcDiagnostic;
     int_T       maskedZcDiagnostic;
     boolean_T   isOutputMethodComputed;
-    int_T       reservedInt;
+    int_T       maxZcBracketingIterations;
```
Comment (Pop0097) [inline Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h:120]: Pretty cool how you're using general data types. Will need to learn more about them

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    IMU_Data_t imuStruct;
+
+    if (resultsDirIsEmpty())  // Data is not ready until we do a step of the model.But we cannot do a step of the model until the state machine gives it instructions.But the state machine cannot give it instructions until it knows where the aircraft is, so this is the starting point.
+    {
+
+        imuStruct.accx = 0;
+        imuStruct.accy = 0;
+        imuStruct.accz = -9.8;
+
+        imuStruct.gyrx = 0;
```
Comment (Pop0097) [inline Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:100]: chuckles, Infinion is in danger

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    dataFile.open(fileName);
+
+    do
+    {
+        previousLine = line;
+
+        getline(dataFile, line);
+    }
+    while (!line.empty());
```
Comment (Pop0097) [inline Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: This do while loop seems a bit suspicious to me:

with the way it's currently structured, it appears as if we will go straight to the end of our file and return the last entry. 

Am I misunderstanding this? 

http://www.cplusplus.com/reference/string/string/empty/

**Review by Pop0097 — COMMENTED** [review]
It seems good to me overall. Just one thing I am slightly unsure of.

File: Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h (hunk 2 above)
Comment (Fberbari) [reply Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h:120]: not my doing lol, all generated. but yeah, quite creative

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp (hunk 4 above)
Comment (Fberbari) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: the getline function saves its state. It always starts at the first line and iterates until it reaches the end. 

This was thoroughly tested in past versions of the simulation, so I can be sure it works.

Comment (Pop0097) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: Okay, then you have my approval

Comment (Fberbari) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: thank you sir!

**Review by Pop0097 — APPROVED** [review]
Nice

**Pull request merged**